### PR TITLE
write eeprom or name if they are available even config is in flash

### PR DESCRIPTION
**Pull request by elral**
## Description of changes

In addition to PR #327 fully backwards compatibility is now ensured.
Once a config is in EEPROM, this config can still be changed even a config is in flash.
Once a name is in EEPROM, this name can still be changed even a config is in flash.
If no name and no config is in EEPROM, name and config can not be changed anymore.

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -718,9 +728,10 @@ void OnGenNewSerial()
 // ************************************************************
 void storeName()
 {
-    if (!configStoredInFlash()) {
+    if (configStoredInEEPROM() || !configStoredInFlash()) {
         MFeeprom.write_byte(MEM_OFFSET_NAME, '#');
         MFeeprom.write_block(MEM_OFFSET_NAME + 1, name, MEM_LEN_NAME - 1);
+        // MFeeprom.commit() is not required, name is always set before config
```
Comment (DocMoebiuz) [inline src/Config.cpp:734]: better remove unnecessary lines

File: src/Config.cpp [hunk 2]
```diff
@@ -718,9 +728,10 @@ void OnGenNewSerial()
 // ************************************************************
 void storeName()
 {
-    if (!configStoredInFlash()) {
+    if (configStoredInEEPROM() || !configStoredInFlash()) {
```
Comment (DocMoebiuz) [inline src/Config.cpp]: is this case not the case when !configStoredInFlash()?

File: src/Config.cpp [hunk 3]
```diff
@@ -735,7 +746,7 @@ void restoreName()
 void OnSetName()
 {
     char *cfg = cmdMessenger.readStringArg();
-    if (!configStoredInFlash()) {
+    if (configStoredInEEPROM() || !configStoredInFlash()) {
```
Comment (DocMoebiuz) [inline src/Config.cpp]: is this case not the case when !configStoredInFlash()?

File: src/Config.cpp [hunk 4]
```diff
@@ -138,16 +140,24 @@ void OnSetConfig()
...
-        cmdMessenger.sendCmd(kStatus, -1); // last successfull saving block is already NULL terminated, nothing more todo
+    if (configStoredInEEPROM() || !configStoredInFlash()) {
+        bool maxConfigLengthNotExceeded = configLengthEEPROM + cfgLen + 1 < MEM_LEN_CONFIG;
+        if (maxConfigLengthNotExceeded) {
+            // save the received config string including the terminatung NULL (+1) to EEPROM
+            MFeeprom.write_block(MEM_OFFSET_CONFIG + configLengthEEPROM, cfg, cfgLen + 1);
+            configLengthEEPROM += cfgLen;
+            cmdMessenger.sendCmd(kStatus, configLengthEEPROM);
+        } else
+            // staus message to connector, failure on setting config
```
Comment (DocMoebiuz) [inline src/Config.cpp:150]: With these multilines in the else block better use `{` `}`

**Review by DocMoebiuz — CHANGES_REQUESTED** [review]
only small notes and a question which use case that is where it can be 

`if (configStoredInEEPROM() || !configStoredInFlash())`

File: src/Config.cpp (hunk 1 above)
Comment (elral) [reply src/Config.cpp:734]: I see this more as an important comment for two reasons:
* users which are not familiary with the code may ask why the name gets not commited to the EEPROM
* should the connector differ between saving the config the name (I don't see a reason to save always the name even it is unchanged) in the future, commiting has to be done. This comment is to remember (hopefully) this.

File: src/Config.cpp [hunk 5]
```diff
@@ -138,16 +140,24 @@ void OnSetConfig()
...
+            // connector does not check for status = -1
+            cmdMessenger.sendCmd(kStatus, -1);
```
Comment (elral) [inline src/Config.cpp:152]: For now the connector reports a successfull upload of the config, even if `-1` is send back.
If no message is send back, the connector shows an error message (due to time out) that the config uploads fails.

Does it make sense to uncomment `cmdMessenger.sendCmd(kStatus, -1);` as long as the connector does not handle this?
There were discussion before to implement a `kError` message which would be sensefull also in this case ;)

**Review by DocMoebiuz — APPROVED** [review]

**Pull request merged**